packages/fxa-settings/src/models/hooks.ts

```diff
@@ -104,13 +105,29 @@ export function useIntegration() {
   }, [clientInfoState, productInfoState]);
 }
 
-export function useExperiments() {
-  const { experiments } = useContext(AppContext);
-  const { Features, Enrollments } = experiments;
-  if (!Features || !Enrollments) {
-    return {};
-  }
-  return { features: Features, enrollments: Enrollments };
+/**
+ * A hook to provide the Nimbus experiments within components.
+ * This hook does not perform a network request.
+ *
+ * @returns the {@link NimbusResult} with experiement information.
+ */
+export function useExperiments(): NimbusResult | null {
+  const { experiments: experimentInfo } = useContext(AppContext) as {
+    experiments: Promise<any>;
+  };
+  const [experiments, setExperiments] = useState<null | NimbusResult>(null);
+  useEffect(() => {
+    experimentInfo.then((exp) => {
+      // Today, we don't need everything from the response so let's only add them as needed.
+      // We map out the response from the doc examples here:
+      // https://github.com/mozilla/experimenter/blob/main/cirrus/README.md
+      setExperiments({
+        features: exp.Features,
+        nimbusUserId: exp.Enrollments.nimbus_user_id || '',
+      } as NimbusResult);
```

Comment:
This mapping is also to remove the confusing capitalization.

Context: https://github.com/mozilla/fxa/pull/18524#discussion_r1990523832